pkg/dockerclient/shell.go

```diff
@@ -358,25 +358,32 @@ func (c *ShellClient) AwaitContainerHealth(containerID string, timeout *time.Dur
 		inspectInterval := 100 * time.Millisecond
 
 		for !timedOut {
-
-			// inspect the container's health, return if it's healthy
-			runResult, err := c.runCommand(nil, "docker inspect --format '{{json .State.Health.Status}}' %s", containerID)
-			if err == nil {
-				stdoutLines := strings.Split(runResult.Output, "\n")
-				lastStdoutLine := c.getLastNonEmptyLine(stdoutLines, 0)
-
-				if lastStdoutLine == `"healthy"` {
+			containers, err := c.GetContainers(&GetContainerOptions{
+				ID:      containerID,
+				Stopped: true,
+			})
+			if err == nil && len(containers) > 0 {
+				container := containers[0]
+
+				// container is healthy
+				if container.State.Health.Status == "healthy" {
 					containerHealthy <- nil
 					return
 				}
-			}
 
-			// wait a bit before retrying
-			c.logger.DebugWith("Container not healthy yet, retrying soon",
-				"timeout", timeout,
-				"containerID", containerID,
-				"inspectOutput", runResult.Output,
-				"nextCheckIn", inspectInterval)
+				// container exited, bail out
+				if container.State.Status == "exited" {
+					containerHealthy <- errors.Errorf("Container exited with status: %d", container.State.ExitCode)
+					return
+				}
+
```

Comment:
perhaps worth handling "dead" status as well

Reply:
what do you mean?

Reply:
in some scenario the container may be in a “dead” state..so maybe worth returning an error on that case too
https://success.docker.com/article/what-is-the-difference-between-dead-and-exited-containers